python/vcastDataServer.py

```diff
@@ -25,27 +25,28 @@
 
 def init_application(logFilePath):
     app = Flask(__name__)
-
     with app.app_context():
 
-        @app.route("/ping")
+        @app.route("/ping", methods=["POST"])
         def pingRoute():
             return ping()
 
-        @app.route("/shutdown")
+        @app.route("/shutdown", methods=["POST"])
         def shutdownRoute():
             return shutdown()
 
-        @app.route("/runcommand")
+        @app.route("/runcommand", methods=["POST"])
         def runcommandRoute():
-            clientRequestText = request.args.get("request")
+            clientRequestText = request.data.decode(
+                "utf-8"
```

Comment:
I wanted to keep the rest of the logic (e.g., decodeRequest(), etc.) unchanged, so I’m still sending the request as a string, like before. However, I can’t use `request.args.get("request")` with Axios because Axios sends its data in the request body, whereas the previous approach retrieved data from the query string `(http:...../?request`). I get this error 


`Enviro server error: client request was improperly formatted, ignoring: 'None'`

When doing: 

`clientRequestText = request.args.get("request")`

Reply:
What's the type of `request.data`? I'm not questioning the removal of the `get`, I'm questioning why you need the `decode`.

Reply:
Can we use `data = request.get_json()`, and then `decodeRequest` doesn't need to do the `json.loads`?

Reply:
Sure, we can do that. I just didn’t want to modify code where I wasn’t specifically asked to, or change the request format to JSON before it's wrong that I changed it.

Reply:
Changed request format in c9c845a8b485173cc75c581d64e8f10b58eeb68e.